vertx-circuit-breaker/src/main/java/io/vertx/ext/circuitbreaker/CircuitBreaker.java

```diff
@@ -0,0 +1,159 @@
+ *
+ * @author <a href="http://escoffier.me">Clement Escoffier</a>
+ */
+@VertxGen
+public interface CircuitBreaker {
+
+  /**
+   * Creates a new instance of {@link CircuitBreaker}.
+   *
+   * @param name    the name
+   * @param vertx   the Vert.x instance
+   * @param options the configuration option
+   * @return the created instance
+   */
+  static CircuitBreaker create(String name, Vertx vertx, CircuitBreakerOptions options) {
+    return new CircuitBreakerImpl(name, vertx, options);
+  }
+
+  /**
+   * Creates a new instance of {@link CircuitBreaker}, with default options
+   *
+   * @param name  the name
+   * @param vertx the Vert.x instance
+   * @return the created instance
+   */
+  static CircuitBreaker create(String name, Vertx vertx) {
+    return new CircuitBreakerImpl(name, vertx, new CircuitBreakerOptions());
+  }
+
+  /**
+   * Closes the circuit breaker. It stops sending events on its state on the event bus.
+   * This method is not related to the {@code close} state of the circuit breaker. To set the circuit breaker in the
+   * {@code close} state, use {@link #reset()}.
+   */
+  void close();
+
+  /**
+   * Sets a {@link Handler} invoked when the circuit breaker state switches to open.
+   *
+   * @param handler the handler, must not be {@code null}
+   * @return the current {@link CircuitBreaker}
+   */
+  CircuitBreaker openHandler(Handler<Void> handler);
+
+  /**
+   * Sets a {@link Handler} invoked when the circuit breaker state switches to half-open.
+   *
+   * @param handler the handler, must not be {@code null}
+   * @return the current {@link CircuitBreaker}
+   */
+  CircuitBreaker halfOpenHandler(Handler<Void> handler);
+
+  /**
+   * Sets a {@link Handler} invoked when the circuit breaker state switches to close.
+   *
+   * @param handler the handler, must not be {@code null}
+   * @return the current {@link CircuitBreaker}
+   */
+  CircuitBreaker closeHandler(Handler<Void> handler);
+
+  /**
+   * Sets a {@link Handler} invoked when the bridge is open to handle the "request".
+   *
+   * @param handler the handler, must not be {@code null}
+   * @return the current {@link CircuitBreaker}
+   */
+  CircuitBreaker fallbackHandler(Handler<Void> handler);
+
+  /**
+   * Resets the circuit breaker state (number of failure set to 0 and state set to closed).
+   *
+   * @return the current {@link CircuitBreaker}
+   */
+  CircuitBreaker reset();
+
+  /**
+   * Explicitly opens the circuit.
+   *
+   * @return the current {@link CircuitBreaker}
+   */
+  CircuitBreaker open();
+
+  /**
+   * @return the current state.
+   */
+  CircuitBreakerState state();
+
+  /**
+   * @return the current number of failures.
+   */
+  long failureCount();
+
+  /**
+   * Executes the given code with the control of the circuit breaker.
+   *
+   * @param code the code
+   * @return the current {@link CircuitBreaker}
+   */
+  CircuitBreaker executeSynchronousBlock(Handler<Void> code);
```

Comment:
Shouldn't the method also take a Handler<AsyncResult> so a result can be returned to the user? (like we do in executeBlocking). Otherwise what happens if an exception is thrown from the blocking code?